src/opm/parser/eclipse/Deck/DeckItem.cpp

```diff
@@ -294,6 +294,16 @@ void DeckItem::push_backDimension( const Dimension& active,
         const bool dim_inactive = du.empty()
             || this->defaultApplied( du.size() - 1 );
 
+        // The data model when it comes to UDA values, dimensions for vectors
+        // and so on is stretched beyond the breaking point. It is a *really*
+        // hard assumption here that UDA values only apply to scalar values.
+        if (du.size() > 1)
+            throw std::logic_error("Internal program meltdown - we do not handle non-scalar UDA values");
+
+        // This is just so broken ....
```

Comment:
This stands out as problematic. The comment only makes me afraid... I think you should add a comment explaining why it is broken, and what consequences it has. It's ok that the code has limitation, as long as we have some understanding of what they are.

Reply:
Comment updated.